launcher.py

```diff
@@ -0,0 +1,335 @@
+import os
+from pathlib import Path
+import subprocess
+import tempfile
+from datetime import datetime
+import torch
+import argparse
+import json
+from jinja2 import Template
+
+from nanotron.logging import human_format
+
+from nanotron.config import (
+    Config,
+    get_config_from_file,
+)
+
+def count_subdirectories(path):
+    return sum(os.path.isdir(os.path.join(path, item)) for item in os.listdir(path))
+
+def launch_slurm_job(launch_file_contents, *args):
+    """
+        Small helper function to save a sbatch script and call it.
+    Args:
+        launch_file_contents: Contents of the sbatch script
+        *args: any other arguments to pass to the sbatch command
+
+    Returns: the id of the launched slurm job
+
+    """
+    with tempfile.NamedTemporaryFile("w") as f:
+        f.write(launch_file_contents)
+        f.flush()
+        return subprocess.check_output(["sbatch", *args, f.name]).decode("utf-8").split()[-1]
+
+def set_nested_attribute(obj, path, value):
+    parts = path.split('.')
+    for part in parts[:-1]:
+        if not hasattr(obj, part):
+            setattr(obj, part, type('', (), {})()) 
+        obj = getattr(obj, part)
+    setattr(obj, parts[-1], value)
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config-path", help="path to the configuration file", type=str, default=None)
+    parser.add_argument("--base-config", help="base config to use", type=str, default=None)
+    parser.add_argument("--run", help="name of the run", type=str, required=True)
+    parser.add_argument("--logs-path", help="path to the logs folder", type=str, default="logs")
+    parser.add_argument("--override", nargs="+", metavar="KEY=VALUE",
+                        help="Override config values. Use dot notation for nested keys.")
+    parser.add_argument("--slurm", action="store_true", help="Launch the job on Slurm")
+    parser.add_argument("--nodes", type=int, help="Number of nodes to use for the job")
+    args = parser.parse_args()
+
+    supported_base_configs = {
```

Comment:
I am not sure I understand the need to have a dict of key to path here. Can't we just specify in command line the path to the yaml file directly ? This will updating `launcher.py` everytime we want to use a different config file

Reply:
It was because i find more simple for the user to do --base-config <name> than --config-path <path_to_config> when using often one config. But tbh i'm not convinced so will remove it